NAS-102188: Rework dataset delete instructions and add intro entry (#1202)
freenas/freenas-docs: userguide/storage.rst

**Edit Permissions:** refer to :ref:`Setting Permissions` for more
information about permissions.

**Delete Dataset:** removes the dataset, snapshots of that dataset, and
any objects stored within the dataset. To remove the dataset, set
:guilabel:`Confirm`, click :guilabel:`DELETE DATASET`, enter the name of

> the dataset that will be deleted, and click :guilabel:`DELETE`.

Reviewer: ```suggestion
the dataset to be deleted, and click :guilabel:`DELETE`.
```

> .. danger:: Removing a dataset is a permanent action and results in

Reviewer: The warning should come before the section telling them how to do it.

> .. danger:: Removing a dataset is a permanent action and can result in

Reviewer 2: Might be a question for Warren, but do we want to be extra sure here? That is, indicate that it surely causes data loss...?

```suggestion
.. danger:: Removing a dataset is a permanent action and results in
```
